src/compiler/crystal/macros/methods.cr

```diff
@@ -1704,6 +1706,15 @@ module Crystal
       end
     end
 
+    def self.class_vars(type)
+      ArrayLiteral.map(type.all_class_vars) do |name, ivar|
```

Comment:
There is difference in the design that should be kept aligned: `instance_vars` performs de check of `InstanceVarContainer` and not in the `all_instance_vars`.